src/main/java/org/simulator/sbml/EquationSystem.java

```diff
@@ -675,9 +680,14 @@ public void init(boolean renewTree, double defaultSpeciesValue, double defaultPa
             for (i = 0; i < model.getConstraintCount(); i++) {
                 if (model.getConstraint(i).isSetMath() && constraintRoots.get(i).compileBoolean(astNodeTime)) {
                     ConstraintEvent evt = new ConstraintEvent(model.getConstraint(i), 0d);
-                    for (ConstraintListener listener : listOfConstraintListeners) {
-                        listener.processViolation(evt);
+                    if (model.getConstraint(i).getUserObject(CONSTRAINT_VIOLATION_LOG) == Boolean.FALSE) {
+                        for (ConstraintListener listener: listOfConstraintListeners) {
+                            listener.processViolation(evt);
+                        }
+                        model.getConstraint(i).putUserObject(CONSTRAINT_VIOLATION_LOG, Boolean.TRUE);
                     }
+                } else {
+                    model.getConstraint(i).putUserObject(CONSTRAINT_VIOLATION_LOG, Boolean.FALSE);
                 }
```

Comment:
We should modify this again. The `ConstraintEvent` contains an instance of the current `Constraint` object. Therefore, the listener can itself push user objects and check their content. We should have done this right away. To safe the effort for initialization, the `ConstraintListener` implementation could treat a missing user object as `false` and just set it when first needed.